Automattic/pocket-casts-ios
# Onboarding: Add plus modal IAP purchases

| 📘 Project: #500 | 🛫 Depends on: #506 |
|:---:|:---:|

This adds real calls to the IAPHelper to make purchase and reacting to the real purchasing states. This logic mirrors the same logic that exists in the ConfirmPaymentViewController.

## To test
#### Setup
1. Open the Xcode Project
3. Hit <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>,</kbd> to open the Scheme editor
4. Click the options tab
5. Select the Pocket Casts Configuration.storekit file in the StoreKit configuration dropdown
6. In Xcode go to Menu Bar > Debug > Store Kit > Manage Transactions

#### Loading prices
1. In Xcode open AppDelegate.swift
2. Comment out line 78: `IapHelper.shared.requestProductInfo()`
3. Open the PlusCoordinator
4. Change line 36 to: `DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(1)) { IapHelper.shared.requestProductInfo() }`
7. Launch the app
10. Go to the Podcasts tab
11. Tap on the create folder button 
12. Tap on the Unlock All Features button
13. ✅ You see a loading spinner, and then the modal is presented with the prices loaded

#### Testing Success
7. Launch the app
9. Sign into an account if you're not
10. Go to the Podcasts tab
11. Tap on the create folder button 
12. Tap on the Unlock All Features button
13. Select the Monthly option
15. Tap the Subscribe button
16. ✅ Verify you see a loading indicator in the button
17. Tap the Subscribe button on the Apple prompt
18. Tap Ok on the "You're all set" alert
19. ✅ Verify you are brought to the "Account Upgraded" screen
20. ✅ Go to the Manage Transactions window
21. ✅ Verify you see the Monthly purchase there
22. Delete it by highlighting it and hitting delete
23. Repeat the steps using the Yearly option

#### Testing failure
1. In Xcode open the Pocket Casts Configuration.storekit file
2. In the menubar go to Editor > Fail Transactions > Select any except Unknown
6. Repeat the steps from above
7. ✅ Verify when you tap the Subscribe button you see an error message appear
8. ✅ Verify the button now says Try Again
9. ✅ Verify the view has been updated to only show your selected option

## Checklist
